database/init.sql

```diff
@@ -21,7 +21,7 @@ CREATE TABLE monitor (
 
 DROP TABLE IF EXISTS run;
 
-CREATE TYPE states AS ENUM ('started', 'completed', 'failed', 'unresolved', 'no_start', 'solo_completed', 'missed');
+CREATE TYPE states AS ENUM ('started', 'completed', 'failed', 'unresolved', 'no_start', 'solo_completed', 'missed', 'solo_missed');
```

Comment:
I thought if no start ping came in, then we never created a job for that end ping, so there wouldnt be some timer that changes 'no_start' to 'missed' in the case of both a start and end ping not coming in?

Reply:
Yes this is true. I may have interpreted things incorrectly but the way I have implemented things is like this:
- a missed start ping creates a run which has state "missed"
- an end ping without a start ping creates a run which has state "no_start"
- a run with state "missed" can never be changed
- a run with "no_start" can be changed to "completed" if the start ping arrives later (this will be included in the start ping if statement in the `pings` controller I am working on now)

The reason we have done it like this is so that we don't have to attempt to match up a late start ping with a "missed" run which will have no run token.

Reply:
Okay I see, that makes sense, I'm still unclear on how would a 'solo_missed' and a 'missed' be different?

Reply:
Yeah I just added that in case we wanted to display a different message to the user. Something like "your job did not start by x" vs "your job did not complete by x". I think there's no difference functionally.

Reply:
Ahh I see okay!